genesis/process/sovereignGenesisBlockCreator.go

```diff
@@ -277,3 +317,53 @@ func setSovereignStakedData(
 
 	return stakingTxs, nil
 }
+
+func createSovereignGenesisESDTTransfers(args ArgsGenesisBlockCreator, scrProcessor process.SmartContractResultProcessor) (map[string]data.TransactionHandler, error) {
+	initialESDTTxs := make(map[string]data.TransactionHandler, 0)
+	initialAccounts := args.AccountsParser.InitialAccounts()
+
+	tokenID := args.Config.SovereignConfig.GenesisConfig.NativeESDT
+	for nonce, initialAcc := range initialAccounts {
+		accInitialBalance := initialAcc.GetBalanceValue()
+
+		log.Debug("creating genesis initial esdt balance",
+			"address", initialAcc.GetAddress(), "balance", accInitialBalance.String(), "tokenID", tokenID)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   uint64(nonce),
+			RcvAddr: initialAcc.AddressBytes(),
+			SndAddr: core.ESDTSCAddress,
+			Data:    createGenesisSCRData(tokenID, accInitialBalance),
+			Value:   big.NewInt(0),
+		}
+
+		retCode, err := scrProcessor.ProcessSmartContractResult(scr)
+		if err != nil {
+			return nil, err
+		}
+
+		if retCode != vmcommon.Ok {
+			log.Error("could not generate initial esdt balances",
+				"func", "createSovereignGenesisESDTTransfers", "ret code", retCode, "address", initialAcc.GetAddress())
+			return nil, errCouldNotGenerateInitialESDTTransfers
+		}
+
+		hash, err := core.CalculateHash(args.Core.InternalMarshalizer(), args.Core.Hasher(), scr)
+		if err != nil {
+			return nil, err
+		}
+
+		initialESDTTxs[string(hash)] = scr
+	}
+
+	return initialESDTTxs, nil
+}
+
+func createGenesisSCRData(tokenID string, value *big.Int) []byte {
+	numTokensToTransferBytes := big.NewInt(1).Bytes()
+	return []byte(core.BuiltInFunctionMultiESDTNFTTransfer +
```

Comment:
Why using `MultiESDTNFTTransfer` for only one ESDT? Not sure what is the reason and if it's more expensive than `ESDTTransfer`, is just an observation.
If you are using multi esdt transfer, maybe you could use `NativeESDT` as `array` and refactor all the references.